### PR TITLE
Support for tenant specific Identity Provider Entity IDs 

### PR DESCRIPTION
**Pull request by ravindudsilva**
Note the subject

Related issue - https://github.com/wso2-enterprise/asgardeo-product/issues/4444

### CODE REVIEW

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java [hunk 1]
```diff
@@ -113,6 +113,24 @@ public static IdentityProviderManager getInstance() {
         return instance;
     }
 
+    /**
+     * Retrieves tenantID.
+     *
+     * @param tenantDomain Tenant name
+     * @return tenantId
+     */
```
Comment (somindatommy) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java:121]: Incomplete method comment; missing exception details and missing `.` at the end.
If we are adding a comment make sure to complete that.

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java [hunk 2]
```diff
@@ -113,6 +113,24 @@ public static IdentityProviderManager getInstance() {
...
+    private static int getTenantID(String tenantDomain) throws IdentityProviderManagementException {
+
+        int tenantId;
+        try {
+            tenantId = IdPManagementServiceComponent.getRealmService().getTenantManager().getTenantId(tenantDomain);
```
Comment (somindatommy) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java:126]: We can just return the value. We do not need a variable here.

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java [hunk 3]
```diff
@@ -113,6 +113,24 @@ public static IdentityProviderManager getInstance() {
...
+        } catch (UserStoreException e) {
+            throw new IdentityProviderManagementException(
+                    "Exception occurred while retrieving Tenant ID from Tenant Domain " + tenantDomain, e);
```
Comment (somindatommy) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java:129]: `Error occurred while retrieving tenant Id from tenant domain: ` > WDYT?

**Review by IsuraD — APPROVED** [review]

**Comment by Wathsara** [comment]
Build verified.

image

**Pull request merged**